### ISSUE TITLE
Hide explorer context menu when it's not json or yml file

### ISSUE THREAD

**Issue by snys98** [hilleer/vscode-yaml-plus-json#258]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when having too much items shown in explorer context menu

**Describe the solution you'd like**
Hide explorer context menu when it's not json or yml file

**Describe alternatives you've considered**
getting used to it

**Additional context**
![image](https://github.com/hilleer/vscode-yaml-plus-json/assets/11873100/5df9191d-b985-48d3-a431-aa3b63d80be9)

**Comment by hilleer** [comment]
Hi @snys98 

Thanks for opening this issue!

Note that the options says "json files" and "yaml files" in plural. The option is meant to help you convert multiple files in the folder. Unfortunately there is (or was not at the time of implementing the feature) any option or listeners to tell if there's such files in folder selected, in order to make a decision if those should be shown or not in the menu.

I'll be happy to consider supplying an option to _always_ disable those menu items though if that would be a helpful feature?

**Comment by snys98** [comment]
Thanks for the reply and it do make sense😄
But I do think that selection of **single or multiple json or yaml** would be a perfect and intuitive implementation over select a folder and convert all sub json or yml files.
Feel free to close this, I'm all good with current implementation, just some suggestions. And thanks again for you hard work👍

**Comment by hilleer** [comment]
I suppose that is a personal preference. Personally I often convert a full folder. Imagine a folder of 10 files you want to convert. I don't think it is very intuitive having to select all of them in order to convert 👍 I think it should be possible to disable/enable this with an option.

**Comment by snys98** [comment]
> I suppose that is a personal preference. Personally I often convert a full folder. Imagine a folder of 10 files you want to convert. I don't think it is very intuitive having to select all of them in order to convert 👍 I think it should be possible to disable/enable this with an option or setting, though.

yes, I agree, it's just a personal preference... maybe it's just me or some minority that prefer to bulk select multiple files and do operations... but I can think another scenario that some files in the folder are in yaml format but not have a `.yaml` extension(maybe yml? or even no file extension) and vice versa..... Yeah I can see this is a rare case, more like picking bones inside an egg, just ignore me hahaha

**Comment by hilleer** [comment]
> > I suppose that is a personal preference. Personally I often convert a full folder. Imagine a folder of 10 files you want to convert. I don't think it is very intuitive having to select all of them in order to convert 👍 I think it should be possible to disable/enable this with an option or setting, though.
> 
> yes, I agree, it's just a personal preference... maybe it's just me or some minority that prefer to bulk select multiple files and do operations... but I can think another scenario that some files in the folder are in yaml format but not have a `.yaml` extension(maybe yml? or even no file extension) and vice versa..... Yeah I can see this is a rare case, more like picking bones inside an egg, just ignore me hahaha

`.yml` extension _should be_ supported already though 👀 From a technical point of view it should be possible to convert any file with either json/yaml content regardless of file extension. It is an interesting take that I have not considered. I guess I expected commonly one would actually use file extensions 🤔
